src/plugins/syscalls/syscalls.cpp

```diff
@@ -106,32 +106,63 @@
 #include <glib.h>
 #include <inttypes.h>
 #include "syscalls.h"
+#include "scproto.h"
 
 static event_response_t cb(drakvuf_t drakvuf, drakvuf_trap_info_t *info) {
+    int i,j;
 
-    syscalls *s = (syscalls*)info->trap->data;
+    syscall_wrapper_t *wrapper = (syscall_wrapper_t *)g_malloc(sizeof(syscall_wrapper_t));
+    wrapper = (syscall_wrapper*)info->trap->data;
```

Comment:
..as you are just overwriting the pointer here, leading to the allocated memory to be lost and leaked.